cosmwasm/contracts/crosschain-registry/src/execute.rs

```diff
@@ -590,93 +867,129 @@ mod tests {
                     (&"osmosis".to_string(), &"cosmos".to_string())
                 )
                 .unwrap(),
-            "channel-0"
+            ("channel-0".to_string(), true)
+        );
+        assert_eq!(
+            CHANNEL_ON_CHAIN_CHAIN_MAP
+                .load(
+                    &deps.storage,
+                    (&"channel-0".to_string(), &"osmosis".to_string())
```

Comment:
`&"channel-0".to_string()` == `"channel-0"` (because the type of "channel-0" is already &str, doing to_string() makes it a String, and taking a reference to it back into &str)

Reply:
https://github.com/osmosis-labs/osmosis/pull/4553/commits/622fe652f5cda595a322d621cb38e689ba2294ca